### ISSUE TITLE
SDL Mixer crash during exit

### ISSUE THREAD

**Issue by wattoc** [haikuports/haikuports#5847]
Apps using SDL Mixer to play midi may crash during exit due to some processes not stopping fully.

I believe the crashes are due to the declaration of BMidiSynth synth in Native_Midi_Haiku.cpp - it appears this doesn't get disposed correctly (I'm not even sure how it is being instantiated properly either) when apps terminate which means it doesn't trigger the Unload() in BSoftSynth.

I'm not sure if this impacts SDL Mixer 2, since I'm only testing against 1.2.  I believe SDL Mixer 2 is using Fluidsynth which doesn't load correctly and crashes anyway - probably worth its own issue.

I've made some local changes to instantiate BMidiSynth during native_midi_detect(), and also added a delete when music is being closed down and the crashes have seemingly gone away.

Happy to take some direction here, still quite new with the Haiku scene.

**Comment by wattoc** [comment]
No longer crashing after #5863 merged in.
